pkg/daemon/controller/controller.go

```diff
@@ -295,6 +304,12 @@ func (sc *ServiceController) reconcileEnvVarsAndCerts(envVars map[string]string,
 	node core.Node) (bool, error) {
 	certsUpdated, err := certs.Reconcile(sc.caBundle)
 	if err != nil {
+		var fileIOErr *certs.FileIOError
+		if errors.As(err, &fileIOErr) {
```

Comment:
nit: errors.As is being used here when the functionality of errors.Is would suffice. This threw me off as everywhere else you use errors.Is